### ISSUE TITLE
Threads leaking on exceptions

### ISSUE THREAD

**Issue by g-gagnon** [celluloid/celluloid#720]
I am currently working on a TCP server using JRuby 9.0.5.0 and Celluloid 0.17.3, where some hardware are connecting to. While working on my load testing, I have an unrelated exception occuring, and I try to terminate the current actor.

```
rescue StandardError => e
      Raven.capture_exception(e)
      terminate unless current_actor.dead?
```

Each failing client tries to reconnect, thus creating a loop of exceptions. Many actors are created, and are supposed to be dying, but the number of active threads keep raising until I reach my system limits.

Is it linked with #716 ?

**Comment by digitalextremist** [comment]
I wouldn't be surprised if the `.dead?` call is raising an exception. But can you post a more complete gist?

**Comment by digitalextremist** [comment]
Bump -- repro? #716 will be resolved in `0.17.4`

**Comment by tarcieri** [comment]
Closing as stale. If you are still interested in this issue, please reopen it.
